tests/app/service/test_api_key_endpoints.py

```diff
@@ -30,9 +30,11 @@ def test_api_key_should_create_new_api_key_for_service(notify_api, notify_db_ses
             )
             assert response.status_code == 201
             assert 'data' in json.loads(response.get_data(as_text=True))
-            saved_api_key: ApiKey = notify_db_session.session.scalar(
-                select(ApiKey).where(ApiKey.service_id == service.id)
-            )
+
+            saved_api_keys: ApiKey = get_model_api_keys(service.id)
+            assert len(saved_api_keys) == 1
+
+            saved_api_key = saved_api_keys[0]
             assert saved_api_key.service_id == service.id
             assert saved_api_key.name == 'some secret name'
             assert saved_api_key.expiry_date is not None
```

Comment:
Is the teardown after these asserts still necessary?  If so, do the asserts need to be wrapped in a "try" to ensure the teardown executes?

Reply:
I tried running this test file with all the teardown stuff removed and everything seemed to work fine without it. Checking the db after the tests showed no artifacts in the related tables (api_keys & api_key_history). So it might not be necessary.
It looks like these teardown steps were actually added by this [massive PR](https://github.com/department-of-veterans-affairs/notification-api/pull/1633) that you, Kyle and Michael contributed to.

Reply:
I get it now.  It is necessary because the test creates an api key using the POST route.